Migrate documentation from Wiki to GitHub (#4)
jenkinsci/toolenv-plugin: README.adoc

[[git-plugin]]
= Tool Environment Plugin
:toc: macro
:toc-title:

link:https://ci.jenkins.io/job/Plugins/job/toolenv-plugin/job/master/[image:https://ci.jenkins.io/job/Plugins/job/toolenv-plugin/job/master/badge/icon[Build Status]]
link:https://github.com/jenkinsci/toolenv-plugin/graphs/contributors[image:https://img.shields.io/github/contributors/jenkinsci/toolenv-plugin.svg?color=blue[Contributors]]

> link:https://github.com/jenkinsci/toolenv-plugin/releases/latest[image:https://img.shields.io/github/release/jenkinsci/toolenv-plugin.svg?label=changelog[GitHub release]]

Reviewer: There don't seem to be any releases of the plugin that are recorded as GitHub releases.  There are two tags which could be used to record a GitHub release, but no release has been recorded for those tags.

This svg is a good choice if the plugin will be [GitHub Releases, possibly with release drafter](https://www.jenkins.io/doc/developer/publishing/documentation/#changelogs) to automate the creation of changelogs.  A [video tutorial](https://www.youtube.com/watch?v=PaQsvli92XY) on release drafter in the Jenkins project is also available.
```suggestion
link:https://github.com/jenkinsci/toolenv-plugin/releases/latest[image:https://img.shields.io/github/release/jenkinsci/toolenv-plugin.svg?label=changelog[GitHub release]]
```

Author: I'm the maintainer and have already enabled release drafter on this plugin and added the config in #PR5. Hopefully this release will be the first one using it.

link:https://plugins.jenkins.io/toolenv[image:https://img.shields.io/jenkins/plugin/i/toolenv.svg?color=blue&label=installations[Jenkins Plugin Installs]]

== Introduction

Tool environment plugin lets you use "tools" in unusual ways, such as from shell scripts.

See http://issues.jenkins-ci.org/browse/JENKINS-4532[JENKINS-4532] for motivation.

Some screenshots are better than a long discussion:

. This is what you define in the Jenkins configuration screen: +
image:docs/images/toolenv-01.png[Jenkins configuration screen]
. This is how you use the plugin in a job: +
image:docs/images/toolenv-02.png[Jenkins job configuration screen]
. This is what you get: +
image:docs/images/toolenv-03.png[Jenkins console output]

Compared to
https://plugins.jenkins.io/custom-tools-plugin/[Custom Tools
Plugin] the advantage is that you can use the tool in arbitrary ways,

> not just running an executable from it. In the future these plugins
> should be merged.
> https://plugins.jenkins.io/shared-objects/[SharedObjects
> Plugin] can also be used to achieve this plugin's effect.

Reviewer: The [shared objects plugin](https://plugins.jenkins.io/shared-objects/) has an [unresolved security issue](https://www.jenkins.io/security/advisory/2017-04-10/).  We should not suggest its use here.

I would prefer that we not describe future activities in plugin documentation, especially for a plugin that was last released 7 years ago.
```suggestion
not just running an executable from it.
```

== Release notes

> See https://github.com/jenkinsci/toolenv-plugin/releases[GitHub Release] for more recent versions.

Reviewer: This assumes the plugin will use GitHub releases in the future.  Does the plugin have a maintainer that intends to convert it to use GitHub releases or have pull requests been submitted that would switch it to use GitHub releases?
```suggestion
See https://github.com/jenkinsci/toolenv-plugin/releases[GitHub Release] for more recent versions.
```

[[git-plugin]]
= Tool Environment Plugin
:toc: macro
:toc-title:

link:https://ci.jenkins.io/job/Plugins/job/toolenv-plugin/job/master/[image:https://ci.jenkins.io/job/Plugins/job/toolenv-plugin/job/master/badge/icon[Build Status]]
link:https://github.com/jenkinsci/toolenv-plugin/graphs/contributors[image:https://img.shields.io/github/contributors/jenkinsci/toolenv-plugin.svg?color=blue[Contributors]]
link:https://github.com/jenkinsci/toolenv-plugin/releases/latest[image:https://img.shields.io/github/release/jenkinsci/toolenv-plugin.svg?label=changelog[GitHub release]]
link:https://plugins.jenkins.io/toolenv[image:https://img.shields.io/jenkins/plugin/i/toolenv.svg?color=blue&label=installations[Jenkins Plugin Installs]]

> [#introduction]

Reviewer: asciidoc will automatically create an identifier for heading with the same text as the heading
```suggestion
```

== Introduction

Tool environment plugin lets you use "tools" in unusual ways, such as from shell scripts.

See http://issues.jenkins-ci.org/browse/JENKINS-4532[JENKINS-4532] for motivation.

Some screenshots are better than a long discussion:

> . This is what you define in Hudson's main config screen: +

Reviewer: Refer to Jenkins rather than Hudson
```suggestion
. This is what you define in the Jenkins configuration screen: +
```

> [.confluence-embedded-file-wrapper .image-center-wrapper]#image:docs/images/toolenv-01.png[image]#

Reviewer: Don't need the confluence markers and can help those using screen readers by adding a brief text description of the image
```suggestion
image:docs/images/toolenv-01.png[Jenkins configuration screen]
```

. This is how you use the plugin in a job: +

> [.confluence-embedded-file-wrapper .image-center-wrapper]#image:docs/images/toolenv-02.png[image]#

Reviewer: Same, remove Confluence markers, add text describing image
```suggestion
image:docs/images/toolenv-02.png[Jenkins job configuration screen]
```

. This is what you get: +

> [.confluence-embedded-file-wrapper .image-center-wrapper]#image:docs/images/toolenv-03.png[image]#

Reviewer: ```suggestion
image:docs/images/toolenv-03.png[Jenkins console output]
```

Compared to
https://plugins.jenkins.io/custom-tools-plugin/[Custom Tools
Plugin] the advantage is that you can use the tool in arbitrary ways,
not just running an executable from it. In the future these plugins
should be merged.
https://plugins.jenkins.io/shared-objects/[SharedObjects
Plugin] can also be used to achieve this plugin's effect.

> [#release-notes]

Reviewer: asciidoc will create that identifier without declaring it in the file
```suggestion
```
